### PR TITLE
Add time calculation for each epoch/batch in trianer

### PR DESCRIPTION
**Pull request by amjltc295**
I think it would be nice to know the training time.
This could be seen by changing the logging configure as well, but the delta time may be hard to find if there are too many batches.

### CODE REVIEW

**Comment by SunQpark** [comment]
Hi, @amjltc295 
I agree that printing the training time would be good, but there are some points that can be improved.

<pre>
Train Epoch: 1 [45056/54000 (83%)] Loss: 0.523107 <b>Time: 0.02s</b>
    epoch          : 1
<b>    epoch_time     : 3.8233487606048584</b>
    loss           : 1.0713483376323052
    my_metric      : 0.642586723663522
    my_metric2     : 0.8484682095125786
<b>    val_epoch_time : 2.062150239944458</b>
    val_loss       : 0.2415582835674286
    val_my_metric  : 0.9347896852355072
    val_my_metric2 : 0.9883803215579711
Saving checkpoint: saved/Mnist_LeNet/1026_042054/checkpoint-epoch1.pth ...
Saving current best: model_best.pth ...
</pre>

1. Displaying the **batch time** is not intuitive
2. **epoch time** does not look good

Since it is displayed on each logging steps and just says 'Time', I think the batch time could be easily mistaken as the step time. 

How do you think about this format?
<pre>
Train Epoch: 1 [<b>3.2s/4s eta</b> 45056/54000 (83%)] Loss: 0.523107
</pre>

**Comment by amjltc295** [comment]
Sorry, I don't understand the meaning of `3.2s/4s eta`.  Could you explain it a bit?
I think the purpose of printing time is to let users estimate the total training time and check if there is a bug that slow down the process. It wouldn't be hard to tell the batch time and step time; or we could add change it to `Batch time` or move it inside the`[ ]`.

**Comment by amjltc295** [comment]
`torch.autograd.profiler` is a new thing for me. It some time for me to figure out how it works, but it seems to be a good solution.
